models/topics.model.js

```diff
@@ -1,9 +1,20 @@
-// const data = require("../db/data")
-const db = require("../db/connection")
+const db = require("../db/connection");
+
+exports.checkTopicExists = (topic) => {
+    return db
+        .query("SELECT * FROM topics WHERE slug = $1", [topic])
+        .then(({ rows }) => {
+            if (rows.length === 0) {
+                return Promise.reject({
+                    status: 404,
+                    msg: "404: Topic Not Found",
+                });
+            }
+        });
```

Comment:
As there're a few check...Exists function, it might be a good idea to make a generic checkExists function. Please take a look at advanced error handling in NCNotes